### DISCUSSION TITLE
Command line run options

### DISCUSSION THREAD

**Discussion by david95p** [sepinf-inc/IPED#1622] [Q&A] [answered]
Hello,
I'm trying to schedule some runs via command line. For this purpose I need run-specific configurations in particular custom categories to export files for each run. 
The idea to solve this problem was to pass a custom _confdir_ as argument in the command.

Another problem I encountered was that, when a run fails for example because there's no space left on the device,  the directory created in the tmp dir is not deleted and a lot of disk space is used.
As the naming of the run-specific tmp directory seems random i can't delete it programmatically without being sure it's related to the failed run and not to a running one. 
With the same idea it would be useful to define a  run-specific _indexTemp_  path. 

I tried to define these parameters with the -X option via command line but that doesn't seem to work. 
Is the -X option implemented for this purpose also? Am i missing something? 

System informations:
- Ubuntu 20.04
- IPED 3.18.15

Many thanks

**Comment by wladimirleite** [comment, marked as answer]
> Hello, I'm trying to schedule some runs via command line. For this purpose I need run-specific configurations in particular custom categories to export files for each run. The idea to solve this problem was to pass a custom _confdir_ as argument in the command.

If I understand the idea, I believe using custom profiles would be an option (command line -profile <profN>). And in each custom profiles you can set the categories to export.
I see you are using IPED 3.18.x. It is possible to use custom profiles with that version, but with 4.x you need to set only things you want to change, so probably it would be easier.

**Reply by lfcnassif** [reply]
> Another problem I encountered was that, when a run fails for example because there's no space left on the device, the directory created in the tmp dir is not deleted and a lot of disk space is used.
> As the naming of the run-specific tmp directory seems random i can't delete it programmatically without being sure it's related to the failed run and not to a running one.
> With the same idea it would be useful to define a run-specific _indexTemp_ path.

About this, temp dir data is not cleaned because user may want to use the --continue option to resume processing. If the error was "no space left on device", it's possible to copy the data to another disk before trying to resume. The temp subdirectory used is random, but it is saved in case/iped/data/prevTempDir.txt file (needed by --continue) if you want to always delete it programatically.

> Is the -X option implemented for this purpose also?

No, that's not its goal, it is used by specific modules to set extra configs, it was not designed to work for all configs.

**Reply by david95p** [reply]
Using now version 4.
I create a custom profile directory containing the file CategoriesToExport.txt and that works fine.

Regargind the tmp directory is it enough to define a _indexTemp_ in a _LocalConfig.txt_ file inside the custom profile? Because that doesn't seem to override de default directory.

**Comment by lfcnassif** [comment]
> * IPED 3.18.15

Please also notice 3.x versions are not supported anymore.

**Comment by david95p** [comment]
Thank  you very much for the hints. 

About the version I am using, I am looking forward to upgrade but I've experienced some changes, in particular, similarly to #1433, by exporting the _Calls_ category tha caller's and reciever's phone numbers  and a timestamp are not exoprted anymore in the files, and I've not figured out a way to retrieve these information yet. 
The result of a call file with 4.1.1:
```
ufed:Direction: Outgoing
ufed:Duration: 00:06:15.4950000
ufed:Source: FaceTime
ufed:Status: Established
ufed:VideoCall: True
ufed:decoding_confidence: High
ufed:extractionId: 0
ufed:extractionName: Advanced Logical
ufed:id: XX
ufed:isrelated: False
ufed:source_index: XX
```
The result of a call file with 3.18.15:
```
Message-Date: 2022-05-23T17:43:56Z
Message-To: XX
ufed:CountryCode: XX
ufed:Direction: Outgoing
ufed:Duration: 00:00:01.7230000
ufed:NetworkName: Unknown network
ufed:Status: Established
ufed:VideoCall: False
ufed:decoding_confidence: High
ufed:extractionId: 0
ufed:extractionName: Advanced Logical
ufed:id: XX
ufed:isrelated: False
ufed:source_index: XX
```
As _Message-To_ or _Message-From_ are essential for me an upgrade is no possible yet.

**Reply by wladimirleite** [reply]
> As _Message-To_ or _Message-From_ are essential for me an upgrade is no possible yet.

I believe that the same information is available in the properties below. Please check if you can find these fields and compare them to 3.18.x in your case. 

```
Communication:Date: 
Communication:From:  
Communication:To:  
```

**Reply by david95p** [reply]
I may need to clear up what I'm trying to do.
- I set _enableAutomaticExportFiles_ to _true_ in _LocalConfig.txt_
- Categories I want to export are defined in _CategoriesToExport.txt_
- The files obtained in dir _Exported_ are then processed by my scripts

Based on this approach where should I look up for these properties? 

I tried again to be sure with version 4.1.1 and in the files relative to calls _Communication_ properties are not written. The entire result for a call is as I reported above. I may have some misconfiguration but the "problem" makes me think that as  _Common_ properties where not written on file as in #1433, also _Comunication_ are not written.

**Reply by lfcnassif** [reply]
This seems a bug, and a similar fix to #1433 may be applied. I can look into this in a couple of hours.

**Reply by lfcnassif** [reply]
Just opened and fixed #1628. In about 10min, a fixed snapshot will be generated here:
https://github.com/sepinf-inc/IPED/actions/runs/4669953719

**Reply by david95p** [reply]
Thank you very much!
Communication properties are now available again.
